cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/MaintenanceEventType.java

```diff
@@ -23,7 +23,38 @@
  * </ul>
  */
 public enum MaintenanceEventType {
-  ADD_BROKER, REMOVE_BROKER, FIX_OFFLINE_REPLICAS, REBALANCE, DEMOTE_BROKER, TOPIC_REPLICATION_FACTOR;
+  // Do not change the order of enums. Append new ones to the end.
+  ADD_BROKER((byte) 0),
+  REMOVE_BROKER((byte) 1),
+  FIX_OFFLINE_REPLICAS((byte) 2),
+  REBALANCE((byte) 3),
+  DEMOTE_BROKER((byte) 4),
+  TOPIC_REPLICATION_FACTOR((byte) 5);
+
+  // This id helps with serialization and deserialization of event types
+  private final byte _id;
```

Comment:
I've always thought this was strange as this can be implemented as
```Java
  byte id() {
    return (byte) ordinal();
  }
```

Reply:
This is indeed cleaner -- thanks.